### ISSUE TITLE
`is.na(fact(.))` seems wrong

### ISSUE THREAD

**Issue by jmbarbone** [jmbarbone/mark#69]
Because `NA` values get their own position, last, they don't show up as NA.
Maybe use a setting for `na_as_na`? This is getting closer to just being a faster version of `factor()`

This behavior isn't explicitly documented and not very intentional.

``` r
library(mark)
x <- fact(c(TRUE, FALSE, NA))
x
#> [1] TRUE  FALSE <NA> 
#> Levels: TRUE FALSE <NA>
as.integer(x)
#> [1] 1 2 3
is.na(x)
#> [1] FALSE FALSE FALSE

is.na(x) <- as.integer(x) == which(is.na(levels(x)))
x
#> [1] TRUE  FALSE <NA> 
#> Levels: TRUE FALSE <NA>
as.integer(x)
#> [1]  1  2 NA
```

<sup>Created on 2022-01-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by jmbarbone** [comment]
Might need a bit more though:

`fact` should problem contain a bit more information:

```r
x <- c("a", "b", NA, "b", "c")
fact(x)
```

should be:

```r
struct(
  c(1L, 2L, 4L, 2L, 3L),
  class = c("fact", "factor"),
  levels = c("a", "b", "c", NA),
  unique = c("a", "b", "c", NA), # would matter for non-characters
  na = 4L # easily accessible
)
```

and `fact_na()` should make:

```r
struct(
  c(1L, 2L, NA_integer_, 2L, 3L),
  class = c("fact", "factor"),
  levels = c("a", "b", "c", NA),
  unique = c("a", "b", "c", NA), # would matter for non-characters
  na = 0L
)
```

and probably be as simple as 

```r
fact_na <- function(x) {
  x <- fact(x)
  [x == attr(x, "na")] <- NA_integer_
  attr(x, "na") <- 0L
  x
}
```

then it can be toggled on/off.  `fact()` will just need to contain that additional `na` attribute

**Comment by jmbarbone** [comment]
will be resolved with #162
